devito/operator.py

```diff
@@ -153,6 +153,11 @@ def arguments(self, *args, **kwargs):
         r_args = [f_n for f_n, f in arguments.items() if isinstance(f, SymbolicData)]
         o_vals = OrderedDict([arg for arg in kwargs.items() if arg[0] in r_args])
 
+        # Are any passed symbols composite? i.e. are they composed of other symbols?
+        for sym in o_vals.values():
```

Comment:
So, if we want to deal with the aliasing problem, I think we need to match the children against their "original names" from the original composite object. The original symbol, however, is lost in the line above, since it requires the name we use to match the original symbols. So we probably need to rewrite the entire section a little, so that we can match everything explicitly by name. How about an explicit loop like this (untested):
```
for name, arg in kwargs:
    if name in arguments:
        # Override the original symbol
        o_vals[name] = arg

        original = arguments[name]
        if original.is_Composite:
            for orig, child in zip(original._children, args._children):
            o_vals[orig.name] = child
```